staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -213,8 +270,39 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, actual, err
 	}
 
-	if err := caseSensitiveJsonIterator.Unmarshal(data, obj); err != nil {
-		return nil, actual, err
+	// If the provided decoding mode is strict, first check if there was a duplicate error earlier in the original JSON
+	// or YAML data, and in that case, return it as a DuplicateFieldError. If that's not a problem, decode using the strict
+	// json-iter configuration, and return a possible error as UnknownFieldError.
+	// If non-strict mode just use the "normal" non-strict json-iter decoder.
+	if s.strict {
+		if yamlToJSONError != nil {
```

Comment:
better do `if s.strict && yamlToJSONError != nil { ... } else if strict { ... } else if ... `. Gives less if-then-else depth, easier to read.